sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1444,6 +1444,31 @@ object SQLConf {
     .intConf
     .createWithDefault(20)
 
+  val FAST_HASH_AGGREGATE_MAX_ROWS_CAPACITY_BIT =
+    buildConf("spark.sql.codegen.aggregate.fastHashMap.capacityBit")
+      .internal()
+      .doc("Capacity for the max number of rows to be held in memory by the fast hash aggregate " +
+        "product operator. The bit is not for actual value, but the actual numBuckets is determined " +
+        "by loadFactor (e.g: default bit value 16 , the actual numBuckets is ((1 << 16) / 0.5).")
+      .intConf
+      .checkValue(bit => bit >= 10 && bit <= 30, "The bit value must be in [10, 30].")
```

Comment:
can you explain how to calculate this valid range?

Reply:
I am not sure about the minimum value. The initial configuration is 1. but  I agree with @maropu suggestion. e.g., 2^1, 2^2,these are meaningless.
for the maximum is 30, because the integer bit is 31 and the actual numBuckets in fast hash map are determined by loadFactor and loadFactor is 0.5 too.
code is:
private double loadFactor = 0.5;
private int numBuckets = (int) (capacity / loadFactor);